app/models/reaction.rb

```diff
@@ -264,7 +264,12 @@ def cleanup_array_fields
   def auto_set_short_label
     prefix = creator.reaction_name_prefix
     counter = creator.counters['reactions'].succ
-    self.short_label = "#{creator.initials}-#{prefix}#{counter}"
+    sh_label = self['short_label']
+    return if short_label && !short_label_changed?
+    return if !(creator && counter)
+
+    label_already_exists = Reaction.find_by(short_label: sh_label)
```

Comment:
``` rb
# build and set short label
# if already exists increase counter & and run again
# if label not used , nothing to do

short_label = "#{creator.initials}-#{prefix}#{counter}"
if Reaction.find_by(short_label: short_label)
 update_counter
 autoset_short_label
end

```

Reply:
@PiTrem - this logic causes more problems, unfortunately. The following reaction API tests fail:

https://github.com/ComPlat/chemotion_ELN/blob/3899b7303da99c3a046f80031007e27fc7e70a0a/spec/api/chemotion/reaction_api_spec.rb#L421

https://github.com/ComPlat/chemotion_ELN/blob/3899b7303da99c3a046f80031007e27fc7e70a0a/spec/api/chemotion/reaction_api_spec.rb#L549

![Screenshot 2023-06-23 113834](https://github.com/ComPlat/chemotion_ELN/assets/67055123/c7009721-9464-4ee1-9b36-60751ae1f5db)

I am proposing that my requested logic be kept as it is.